scripts/receive-payjoin.py

```diff
@@ -71,19 +123,21 @@ def receive_payjoin_main(makerclass):
             options.mixdepth) + ", no coins. Shutting down.")
         sys.exit(EXIT_ARGERROR)
 
-    maker = makerclass(wallet_service, options.mixdepth, receiving_amount)
-    
-    jlog.info('starting receive-payjoin')
-    clientfactory = JMClientProtocolFactory(maker, proto_type="MAKER")
+    # the receiving address is sourced from the 'next' mixdepth
+    # to avoid clustering of input and output:
+    next_mixdepth = (options.mixdepth + 1) % (
+        wallet_service.wallet.mixdepth + 1)
+    bip78_receiving_address = wallet_service.get_internal_addr(next_mixdepth)
 
-    nodaemon = jm_single().config.getint("DAEMON", "no_daemon")
-    daemon = True if nodaemon == 1 else False
-    if jm_single().config.get("BLOCKCHAIN", "network") in ["regtest", "testnet"]:
-        startLogging(sys.stdout)
-    start_reactor(jm_single().config.get("DAEMON", "daemon_host"),
-                  jm_single().config.getint("DAEMON", "daemon_port"),
-                  clientfactory, daemon=daemon, p2ep=True)
+    pj_server = PayjoinServer(wallet_service, options.mixdepth,
+                    CCoinAddress(bip78_receiving_address), bip78_amount)
+    site = Site(pj_server)
+    site.displayTracebacks = False
+    jmprint("Attempting to start serving Tor HS on port: " + str(
```

Comment:
"Tor HS" should be renamed to "Onion service" as this how Tor docs refer to them nowadays, see https://community.torproject.org/onion-services/, for example.

Reply:
Related - https://github.com/bitcoin/bitcoin/pull/19638

Reply:
Noted, will do.

Reply:
Done in https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/682/commits/3a6e180bfeacd672d7dd531fdbaeaaef616559d3